examples/cryptocurrency/src/lib.rs

```diff
@@ -207,7 +207,8 @@ pub enum Error {
 
 impl From<Error> for ExecutionError {
     fn from(value: Error) -> ExecutionError {
-        ExecutionError::new(value as u8)
+        let description = format!("{}", &value);
```

Comment:
hmm, why not https://doc.rust-lang.org/std/error/trait.Error.html#tymethod.description ?

Reply:
As far as I understand this is the reason
```
#[derive(Debug, Fail)]
#[repr(u8)]
pub enum Error {
...
}
```
this is not an `std::error::Error`. Of course you can implement the trait, but I worked in the context of the example.